### DISCUSSION TITLE
Click Event on Marker / Custom Popup

### DISCUSSION THREAD

**Discussion by dariokueffer** [maplibre/maplibre-gl-js#2060] [Q&A] [answered]
Hi there

I'm wondering whether it is possible to have a click event on markers. I couldn't find anything in the docs. I'd like to have a sort of custom pupup, which is not bound to a specific marker but rather a general popup window. See the attached image as an example.

Does anyone have an idea how to implement this with maplibre? Thank's a lot! 🫶

<img width="1053" alt="Bildschirm­foto 2023-01-12 um 10 24 04" src="https://user-images.githubusercontent.com/42995703/212029087-5839c5f6-8eb4-4dd5-9385-5bb23cdd29e2.png">

**Comment by wipfli** [comment]
Hi @dariokueffer. Not related to your question, but we will have a small MapLibre Switzerland meetup in Olten on Jan 30th, 2023, at 18:30. Let me know if you would like to join. You find me on slack at https://slack.openstreetmap.us

**Comment by HarelM** [comment, marked as answer]
If the markers are part of the style-spec (not html markers) than you need to `queryRenderFeautre` in the callback of the `map.on('click', ...)` event.
Once you get the feature and some properties you saved with it you can call whatever UI framework you use to create a toast/bottom banner.
If the markers are html markers then this is a "regular" html interaction you need to implement.
Hope that helps.
The following is an example of HTML markers:
https://israelhiking.osm.org.il/
